cdap-data-fabric/src/main/java/co/cask/cdap/store/OwnerTable.java

```diff
@@ -0,0 +1,119 @@
+   * @throws AlreadyExistsException is thrown when the key entityId already exists.
+   */
+  public void add(final NamespacedEntityId entityId,
+                  final KerberosPrincipalId kerberosPrincipalId) throws IOException, AlreadyExistsException {
+    Optional<StructuredRow> row = table.read(ImmutableList.of
+        (Fields.stringField(StoreDefinition.OwnerStore.PRINCIPAL_FIELD, createRowKey(entityId))));
+    if (row.isPresent()) {
+      throw new AlreadyExistsException(entityId,
+          String.format("Owner information already exists for entity '%s'.",
+              entityId));
+    }
+    Field<String> principalField = Fields.stringField(StoreDefinition.OwnerStore.PRINCIPAL_FIELD,
+                                                      createRowKey(entityId));
+    Field<byte[]> keytabField = Fields.bytesField(StoreDefinition.OwnerStore.KEYTAB_FIELD,
+        Bytes.toBytes(kerberosPrincipalId.getPrincipal()));
+    table.upsert(ImmutableList.of(principalField, keytabField));
+  }
+
+  /**
+   * Function that checks if a given entity id exists.
+   *
+   * @param entityId NamespacedEntityId to check for existence.
+   * @return true if the entityId exists, false otherwise.
+   * @throws IOException is thrown when there is an error reading from the table.
+   */
+  public boolean exists(final NamespacedEntityId entityId) throws IOException {
+    Optional<StructuredRow> row = table.read(ImmutableList.of
+        (Fields.stringField(StoreDefinition.OwnerStore.PRINCIPAL_FIELD, createRowKey(entityId))));
+    return row.isPresent();
+  }
+
+  /**
+   * Get Kerberos principal for a given entity id
+   *
+   * @param entityId NamespacedEntityId for the requested KerberosPrincipalId.
+   * @return KerberosPrincipalId for the given entityId.
+   * @throws IOException is thrown when there is an error reading from the table.
+   */
+  @Nullable
+  public KerberosPrincipalId getOwner(final NamespacedEntityId entityId) throws IOException {
+    Optional<StructuredRow> row = table.read(ImmutableList.of
+        (Fields.stringField(StoreDefinition.OwnerStore.PRINCIPAL_FIELD, createRowKey(entityId))));
+
+    return row.isPresent() ?
+        new KerberosPrincipalId(Bytes.toString(row.get().getBytes(StoreDefinition.OwnerStore.KEYTAB_FIELD))) : null;
+  }
+
+  /**
+   * Delete the entityId and corresponding kerberos principalId.
+   *
+   * @param entityId NamespacedEntityId to delete.
+   * @throws IOException is thrown when there is an error deleting the entry.
+   */
+  public void delete(final NamespacedEntityId entityId) throws IOException {
+    table.delete(ImmutableList.of(Fields.stringField(StoreDefinition.OwnerStore.PRINCIPAL_FIELD,
+                                                     createRowKey(entityId))));
+  }
+
+  private static String createRowKey(NamespacedEntityId targetId) {
+    String targetType = EntityIdKeyHelper.getTargetType(targetId);
+    MDSKey.Builder builder = new MDSKey.Builder();
```

Comment:
Is there a reason we still use MDSKey here instead of just having two row keys, targetType and targetId?

Reply:
Addressed comments.